solr/core/src/test/org/apache/solr/core/FSPRSTest.java

```diff
@@ -200,6 +242,66 @@ public void testShardSplit() throws Exception {
 
   }
 
+  public void testPRSCollectionCreation() throws Exception {
+    String COLL = "prs_test_coll";
+    MiniSolrCloudCluster cluster =
+        configureCluster(1)
+            .withJettyConfig(jetty -> jetty.enableV2(true))
+            .addConfig("conf", configset("conf2"))
+            .configure();
+
+    System.setProperty("overseer.node", "true");
+    JettySolrRunner overseerNode = cluster.startJettySolrRunner();
+    System.clearProperty("overseer.node");
+
+    JettySolrRunner leader = getOverseerleader(cluster);
+    Thread.sleep(8000);
+    assertEquals(overseerNode, leader);
+
+    CoreContainer dataCore = cluster.getJettySolrRunner(0).getCoreContainer();
+    OverseerListener dataNodeOverseerListener = new OverseerListener();
+    dataCore.getZkController().getOverseer().setTestOverseerMessageListener(dataNodeOverseerListener);
+    PrsUpdateListener dataNodePrsListener = new PrsUpdateListener();
+    dataCore.getZkController().zkStateReader.setTestPerReplicaStateUpdateListener(dataNodePrsListener);
+
+    CoreContainer overseerCore = leader.getCoreContainer();
+    OverseerListener overseerNodeOverseerListener = new OverseerListener();
+    overseerCore.getZkController().getOverseer().setTestOverseerMessageListener(overseerNodeOverseerListener);
+    PrsUpdateListener overseerNodePrsListener = new PrsUpdateListener();
+    overseerCore.getZkController().zkStateReader.setTestPerReplicaStateUpdateListener(overseerNodePrsListener);
+
+    CollectionAdminRequest.createCollection(COLL, "conf", 1, 1)
+              .setPerReplicaState(Boolean.TRUE)
+              .process(cluster.getSolrClient());
+    cluster.waitForActiveCollection(COLL, 1, 1);
+
+    assertTrue("overseer node should not update prs state " + overseerNodePrsListener, overseerNodePrsListener.events.size() == 1);
+    assertTrue("data node should have 3 prs state update " + dataNodePrsListener , dataNodePrsListener.events.size() == 3);
```

Comment:
Very nice that we can compare the data node update vs os node update now! 💪🏼 

 This has major advantage over my other test `testZkNodeVersions` which checks the znode version and child version but does not know who updates it. 

We can consider removing `testZkNodeVersions` in the future. However, there's one thing that `testZkNodeVersions` catches while this test case does not - there are direct updates on state.json performed by overseer node, which NOT caught by dataNodeOverseerListener as  it does not go through the overseer q 😞  (so it's actually 2 updates to state.json from overseer instead of 1). This is the update : https://github.com/fullstorydev/lucene-solr/blob/release/8.8/solr/core/src/java/org/apache/solr/cloud/api/collections/CreateCollectionCmd.java#L323 . in order to catch it we might need to instrument even lower level (zkclient) as discussed earlier